### PR TITLE
feat: build and push action for ecr and docker

### PR DESCRIPTION
**Pull request by krol3**
Add action to publish in ecr and docker.

Review the docker test [here](https://hub.docker.com/repository/docker/krol/harbor-scanner-trivy) 
Review the public ecr test [here](https://gallery.ecr.aws/v4f3f9r2/harbor-scanner-trivy) 
Review the github action test [here](https://github.com/krolval/harbor-scanner-trivy/runs/1601393072)

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USER }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.aws
+          username: ${{ secrets.ECR_ACCESS_KEY_ID }}
+          password: ${{ secrets.ECR_SECRET_ACCESS_KEY }}
```
Comment (danielpacak) [inline .github/workflows/publish.yml:39]: Hey @krol3 

I think that we could just take these two login steps for DockerHub and ECR and add them to the existing release workflow, which is triggered whenever we create an annotated Git tag (e.g. `git tag -a v1.4.0 -m 'Release v1.4.0'`):
https://github.com/aquasecurity/harbor-scanner-trivy/blob/main/.github/workflows/release.yml#L29

We're also using GoRelease, which builds Docker image and adds metadata labels. It also allows you to add ECR image template in https://github.com/aquasecurity/harbor-scanner-trivy/blob/main/.goreleaser.yml#L29

**Review by danielpacak — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml (hunk 1 above)
Comment (krol3) [reply .github/workflows/publish.yml:39]: Great idea!! I will update the PR

Comment (krol3) [reply .github/workflows/publish.yml]: Follow the test [here](https://github.com/krolval/harbor-scanner-trivy/actions/runs/440965817)

**Review by danielpacak — APPROVED** [review]
LGTM!

**Pull request merged**